### PR TITLE
Send mail async tasks

### PR DESCRIPTION
**Pull request by raressiteavu**
Refs https://helpdesk.eaudeweb.ro/issues/24753

### CODE REVIEW

File: core/api/views/cp_reports.py [hunk 1]
```diff
@@ -387,6 +388,9 @@ def put(self, request, *args, **kwargs):
 
         current_obj.delete()
 
+        if new_instance.status == CPReport.CPReportStatus.FINAL:
+            send_mail_report_submit.delay(new_instance.id)  # send mail to MLFS
```
Comment (sorinstelian) [inline core/api/views/cp_reports.py:392]: We should also test that tasks sending emails get called when needed (use `CELERY_ALWAYS_EAGER` or something similar)

File: core/tasks.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    if cp_comment.comment_type == CPComment.CPCommentType.COMMENT_COUNTRY:
+        recipients = User.objects.filter(user_type=User.UserType.SECRETARIAT)
+    else:
+        recipients = User.objects.filter(
+            user_type=User.UserType.COUNTRY_USER, country=cp_report.country
+        )
+
+    for recipient in recipients:
+        send_mail(
```
Comment (sorinstelian) [inline core/tasks.py:29]: Erm you could send the mail to multiple user, don't think a `for` loop is needed.

File: docker-compose.override.test.yml [hunk 3]
```diff
@@ -15,3 +15,7 @@ services:
     build:
       context: .
       dockerfile: docker/nginx.docker
+
+  celery:
+    profiles:
```
Comment (sorinstelian) [inline docker-compose.override.test.yml:20]: don't you also need to add rabbitmq and redis to the `donotstart` profile?

File: docker/celery.docker [hunk 4]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.11-slim
+
+ARG REQFILE=requirements.txt
+
+ENV PYTHONUNBUFFERED 1
+ENV PIP_NO_CACHE_DIR=1
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1
+
+RUN mkdir /.fs /app
```
Comment (sorinstelian) [inline docker/celery.docker:9]: I'm not sure this `mkdir` is needed if you also define the volume in docker-compose. BTW, did you test that data in this volume written by the app is accessible in the celery container?

File: docker-compose.override.test.yml (hunk 3 above)
Comment (raressiteavu) [reply docker-compose.override.test.yml:20]: I tried in [this](https://github.com/multilateralfund/ors/pull/228/commits/d5890d72d42956be19bd9d79fab6186765cc8699) commit, but the tests will fail.

File: core/api/views/settings.py [hunk 5]
```diff
@@ -52,5 +52,13 @@ def get(self, *args, **kwargs):
...
                 ),
             },
+            "send_mail": config.SEND_MAIL,
         }
         return Response(settings)
+
+    def post(self, request, *args, **kwargs):
+        send_mail = request.data.get("send_mail")
+        if isinstance(send_mail, bool):
+            config.SEND_MAIL = send_mail
```
[the remark this answers is not in the document]
Comment (raressiteavu) [reply core/api/views/settings.py:62]: Yes, I think. Tested [here](https://github.com/multilateralfund/ors/pull/228/files#diff-2e1b53ff3288d55b4dc5ae1f7056a3b5d9d6855fef824d2778eb28427eb23196R28).

File: core/tasks.py (hunk 2 above)
Comment (raressiteavu) [reply core/tasks.py:29]: Yes... The only difference is that all the recipients will be visible in the `To` field.

**Review by sorinstelian — APPROVED** [review]

**Pull request merged**